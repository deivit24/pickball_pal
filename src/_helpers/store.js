import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
