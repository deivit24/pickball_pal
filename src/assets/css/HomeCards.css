#HomeCards {
  margin-bottom: 150px;
  text-align: center;
}

#HomeCards h1 {
  font-weight: 700;
  margin-bottom: 100px;
}
#HomeCards .grid .paper {
  text-align: center;
  padding: 10px;
  margin-bottom: 25px;
}
#HomeCards .grid .paper img {
  width: 64px;
  margin: 25px 0px;
}

#HomeCards .grid .paper h4 {
  font-weight: 700;
  color: #0077ff;
  margin-bottom: 25px;
}

#HomeCards .grid .paper p {
  color: #333;
  padding: 0 20px;
}
