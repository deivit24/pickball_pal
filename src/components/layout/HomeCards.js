// React
import React from 'react';
// React Router DOM
import { Link } from 'react-router-dom';
// @material-ui/core components
/**
 * I tried using a the Material UI Grid System
 * Still ended using Bootstrap along with it
 */
import { Button, Grid, Paper } from '@material-ui/core';
//Home CSS
import '../../assets/css/HomeCards.css';
// Importing Photos
import earth from '../../assets/img/earth.jpg';
import person from '../../assets/img/person.jpg';
import message from '../../assets/img/message.jpg';

const HomeCards = () => {
  return (
    <div id="HomeCards" className="container">
      <h1>Perks of signing up</h1>
      <Grid container className="grid" spacing={3}>
        <Grid item sm={12} md={4}>
          <Paper className="paper">
            <img src={person} alt="" />
            <h4>Create a Profile</h4>
            <p>
              You can create and edit a profile that other players can find. You
              can display your bio, playing style level, social media accounts,
              and where you play.
            </p>
          </Paper>
        </Grid>
        <Grid item sm={12} md={4}>
          <Paper className="paper">
            <img src={message} alt="" />
            <h4>Direct Message Players</h4>
            <p>
              You can send other players direct messages and directly set up
              your own match or practice schedule with the player. We take no
              payment for connecting players.{' '}
            </p>
          </Paper>
        </Grid>
        <Grid item sm={12} md={4}>
          <Paper className="paper">
            <img src={earth} alt="" />
            <h4>Post to the Community</h4>
            <p>
              You can post to the Pickleball community and filter posts by
              location. That way your post doesn't get lost in the world.
              Comment and like other posts as well.{' '}
            </p>
          </Paper>
        </Grid>
      </Grid>
      <Link to="/register">
        <Button variant="contained" color="primary" size="large">
          Create Profile
        </Button>
      </Link>
    </div>
  );
};

export default HomeCards;
