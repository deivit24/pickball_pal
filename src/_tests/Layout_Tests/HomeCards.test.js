import React from 'react';
import { render } from '@testing-library/react';
import HomeCards from '../../components/layout/HomeCards';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing HomeCards

describe('Testing the HomeCards', () => {
  it('it renders HomeCards without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <HomeCards />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <HomeCards />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
