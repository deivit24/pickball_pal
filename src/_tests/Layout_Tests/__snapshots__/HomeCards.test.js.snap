// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the HomeCards matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    id="HomeCards"
  >
    <h1>
      Perks of signing up
    </h1>
    <div
      class="MuiGrid-root grid MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="person.jpg"
          />
          <h4>
            Create a Profile
          </h4>
          <p>
            You can create and edit a profile that other players can find. You can display your bio, playing style level, social media accounts, and where you play.
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="message.jpg"
          />
          <h4>
            Direct Message Players
          </h4>
          <p>
            You can send other players direct messages and directly set up your own match or practice schedule with the player. We take no payment for connecting players. 
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="earth.jpg"
          />
          <h4>
            Post to the Community
          </h4>
          <p>
            You can post to the Pickleball community and filter posts by location. That way your post doesn't get lost in the world. Comment and like other posts as well. 
          </p>
        </div>
      </div>
    </div>
    <a
      href="/register"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Create Profile
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</DocumentFragment>
`;
